scipy/cluster/tests/test_vq.py

```diff
@@ -261,12 +263,8 @@ def test_kmeans2_init(self):
         kmeans2(data, 3, minit='points')
         kmeans2(data[:, :1], 3, minit='points')  # special case (1-D)
 
-        # minit='random' can give warnings, filter those
-        with warnings.catch_warnings():
-            warnings.filterwarnings('ignore',
-                        message="One of the clusters is empty. Re-run")
-            kmeans2(data, 3, minit='random')
-            kmeans2(data[:, :1], 3, minit='random')  # special case (1-D)
+        kmeans2(data, 3, minit='random')
```

Comment:
Is this correct --- the comment says the warning is raised only sometimes? 
Does the test set the random generator seed? (If not, it probably should...)